storage/raft.proto

```diff
@@ -35,9 +36,28 @@ message RaftMessageRequest {
   optional raftpb.Message message = 4 [(gogoproto.nullable) = false];
 }
 
-// RaftMessageResponse is an empty message returned by raft RPCs. If a
-// response is needed it will be sent as a separate message.
+message RaftMessageResponseUnion {
```

Comment:
This is a strange construction. What else do you envision being added to this union? Why not just make the error optional in the `RaftMessageResponse`?